internals/maintenance_scripts.py

```diff
@@ -450,3 +450,52 @@ def get_template_data(self, **kwargs):
 
     return (f'{len(entities_to_write)} Stages updated with trial data.\n'
             f'{extensions_cleared} extension requests cleared.')
+
+class BackfillFeatureEnterpriseImpact(FlaskHandler):
+  def get_template_data(self, **kwargs):
+    """Backfill enterprise_impact firld for all features."""
+    self.require_cron_header()
+    count = 0
+    batch = []
+    BATCH_SIZE = 100
+    feature_ids = set()
+  
+    stages: ndb.Query = Stage.query(Stage.stage_type == STAGE_ENT_ROLLOUT, Stage.archived == False)
+    # Update enterprise_impact to be the highest impact set on any of the rollout steps.
+    for stage in stages:
+      feature_entry = FeatureEntry.get_by_id(stage.feature_id)
+      feature_ids.add(stage.feature_id)
+      if feature_entry == None:
+        continue
+      new_impact = stage.rollout_impact + 1
+      if new_impact <= feature_entry.enterprise_impact:
+        continue  
+      feature_entry.enterprise_impact = new_impact
+      batch.append(feature_entry)
```

Comment:
Using a list of feature entries rather than a set of IDs might cause a problem in the rare case where a feature has two rollout steps that both cause enterprise_impact to be set.   That could append the same feature entry twice to the batch list, which NDB might treat as an error or it could be ambiguous as to which one is written to the datastore last.